sync15-adapter/src/state.rs

```diff
@@ -0,0 +1,640 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+use std::collections::{HashMap, HashSet};
+use std::mem;
+
+use bso_record::BsoRecord;
+use client::SetupStorageClient;
+use collection_keys::CollectionKeys;
+use error::{self, ErrorKind};
+use key_bundle::KeyBundle;
+use record_types::{MetaGlobalEngine, MetaGlobalRecord};
+use request::InfoConfiguration;
+use util::{random_guid, ServerTimestamp, SERVER_EPOCH};
+
+use self::SetupState::*;
+
+const STORAGE_VERSION: usize = 5;
+
+lazy_static! {
+    /// Maps names to storage versions for engines to include in a fresh
+    /// `meta/global` record.
+    static ref DEFAULT_ENGINES: Vec<(&'static str, usize)> = vec![
+        ("passwords", 1),
+        // Unsupported engines.
```

Comment:
None/Any of these may or may not be supported. Passwords we have a separate implementation already but it's not like this code on it's own supports passwords.